File: fyn-flutter-app/lib/features/post/data/models/report_post_request.dart
import 'report_reason.dart';

class ReportPostRequest {
  final ReportReason reason;
  final String? description;

  ReportPostRequest({
    required this.reason,
    this.description,
  });

  Map<String, dynamic> toJson() {
    return {
      'reason': reason.name,
      'description': description,
    };
  }
}

File: fyn-monolithic/src/main/java/com/fyn_monolithic/dto/request/post/ReportPostRequest.java
package com.fyn_monolithic.dto.request.post;

import com.fyn_monolithic.model.post.ReportReason;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ReportPostRequest {
    @NotNull
    private ReportReason reason;
    private String description;
}
